ecommerce/extensions/payment/processors/payu.py

```diff
@@ -0,0 +1,292 @@
+ProductClass = get_model('catalogue', 'ProductClass')
+Source = get_model('payment', 'Source')
+SourceType = get_model('payment', 'SourceType')
+
+
+class Payu(BasePaymentProcessor):
+    """
+    PayU payment processor (November 2016)
+
+    For reference, see
+    http://developers.payulatam.com/en/web_checkout/integration.html
+    """
+
+    NAME = 'payu'
+
+    def __init__(self):
+        """
+        Constructs a new instance of the PayU processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor
+            AttributeError: If LANGUAGE_CODE setting is not set.
+        """
+        configuration = self.configuration
+        self.payment_page_url = configuration['payment_page_url']
+        self.merchantId = configuration['merchant_id']
+        self.accountId = configuration['account_id']
+        self.ApiKey = configuration['api_key']
+        self.tax = configuration['tax']
+        self.taxReturnBase = configuration['tax_return_base']
+        self.test = configuration['test']
+        self.responseUrl = configuration['response_url']
+        self.confirmationUrl = configuration['confirmation_url']
+
+    @property
+    def receipt_page_url(self):
+        return get_lms_url(self.configuration['receipt_path'])
+
+    @property
+    def cancel_page_url(self):
+        return get_lms_url(self.configuration['cancel_path'])
+
+    @property
+    def error_page_url(self):
+        return get_lms_url(self.configuration['error_path'])
+
+    def get_transaction_parameters(self, basket, request=None):
+        """
+        Generate a dictionary of signed parameters PayU requires to complete a transaction.
+
+        Arguments:
+            basket (Basket): The basket of products being purchased.
+
+        Keyword Arguments:
+            request (Request): A Request object which could be used to construct an absolute URL; not
+                used by this method.
+
+        Returns:
+            dict: PayU-specific parameters required to complete a transaction, including a signature.
+        """
+        parameters = {
+            'payment_page_url': self.payment_page_url,
+            'merchantId': self.merchantId,
+            'accountId': self.accountId,
+            'ApiKey': self.ApiKey,
+            'referenceCode': basket.order_number,
+            'tax': self.tax,
+            'taxReturnBase': self.taxReturnBase,
+            'currency': basket.currency,
+            'test': self.test,
+            'buyerEmail': basket.owner.email,
+            'buyerFullName': basket.owner.full_name,
+            'amount': str(basket.total_incl_tax),
+            'responseUrl': self.responseUrl,
+            'confirmationUrl': self.confirmationUrl,
+        }
+
+        single_seat = self.get_single_seat(basket)
+
+        if single_seat:
+            parameters['description'] = single_seat.course_id
+        else:
+            parameters['description'] = 'default_test_description'
```

Comment:
The case when there is no single_seat should never happen. Tha basket should always contain a product of type "seat". I will delete the `else` code part